### ISSUE TITLE
HeroLab xpath function errors with values that contain commas

### ISSUE THREAD

**Issue by dbirchbauer** [RPTools/maptool#2229]
**Describe the bug**
Looks like I've just found a possible glitch with XPath. when it returns a stringlist its not properly escaping values with , which means that when you then iterate over it, "Restoration, Lesser" becomes "Restoration", and "Lesser".

**To Reproduce**
use xpath to pull variables IE spell names that contain a ","  
```
[h: spells = herolab.XPath('/document/public/character/spellsknown/spell/@name')]
[h: broadcast(spells)]
[h, foreach(spell_name, spells), CODE: {
```
xpath returns `Acid Splash, Barkskin, Bull's Strength, Detect Magic, Enlarge Person, Guidance, Haste, Heroism, Mage Armor, Message, Open/Close, Read Magic, Rejuvenate Eidolon, Lesser, Resist Energy, Restore Eidolon, Restore Eidolon, Lesser, Shield, Unfetter` which then when iterated over breaks results like `Restore Eidolon, Lesser` treating it as `Restore Eidolon` and `Lesser` not as 1 spell

**Expected behavior**
when pulling and then iterating over the results of an xpath fields should not be mangled

**MapTool Info**
- Version: 1.7.0
- Install: New

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 10

**Comment by selquest** [comment]
Looks like the `herolab.Xpath()` function should get an optional `delim` param that can override the default comma and even return json.  Looks like a straightforward change to make, but I don't use Hero Lab - any chance you can provide a file for testing?

**Comment by Phergus** [comment]
Here's a test file Jamz had for HeroLab stuff.  Know idea if it has the right info in it.

[MapTool_Import_Tests.zip](https://github.com/RPTools/maptool/files/5251781/MapTool_Import_Tests.zip)

**Comment by Phergus** [comment]
Any connect to #1562 or #1733?

**Comment by dbirchbauer** [comment]
Just remove the .zip from the attached file, take a look at Babau (Demon) and try to parse the spell like abilities that it has.  "Teleport, Greater" should get mangled 
[PZOPSS1098E - Pathfinder Society Scenario 10-98 - Siege of Gallowspire (7-8).por.zip](https://github.com/RPTools/maptool/files/5252036/PZOPSS1098E.-.Pathfinder.Society.Scenario.10-98.-.Siege.of.Gallowspire.7-8.por.zip)

**Comment by selquest** [comment]
Added `delim` parameter to herolab.XPath, allowing it to return a String List or JsonArray. Default remains `", "`.

Sample output from the Babau creature in the provided portfolio:

**Default String List**: `[r: herolab.XPath("/document/public/character/spelllike/special/@name")]`
```
Darkness (At will), Dispel Magic (At will), See Invisibility (Constant), Summon (level 3, 1 babau 40%, 1/day) (Sp), Teleport, Greater (self plus 50 lbs. of objects only, At will)
```

**Custom-Delimited String List**: `[r: herolab.XPath("/document/public/character/spelllike/special/@name", token.name, "|")]`
```
Darkness (At will)|Dispel Magic (At will)|See Invisibility (Constant)|Summon (level 3, 1 babau 40%, 1/day) (Sp)|Teleport, Greater (self plus 50 lbs. of objects only, At will)
```

**JSON Array**: `[r: herolab.XPath('/document/public/character/spelllike/special/@name', currentToken(), "json")]`
```json
[
  "Darkness (At will)",
  "Dispel Magic (At will)",
  "See Invisibility (Constant)",
  "Summon (level 3, 1 babau 40%, 1/day) (Sp)",
  "Teleport, Greater (self plus 50 lbs. of objects only, At will)"
]
```

For anybody else who hadn't looked at the HeroLab integration in years, http://lmwcs.com/rptools/wiki/How_to_use_HeroLab_Functions may be helpful to you too.  I added the Tutorial category because I had trouble finding it.

@dbirchbauer It's important to note that using the default comma delimiter will still produce errors on this data - MapTool isn't really "mangling" it, it's just that MT's String List convenience type is really only suited for simple elements.  There's no escape mechanism to squash delimiters that appear inside elements, so it's important to choose a delimiter that does not occur inside the text being joined (which wasn't previously possible with this particular function).  When dealing with more complicated text elements, it will usually be better to deal with the json versions instead.
